elyra/pipeline/registry.py

```diff
@@ -0,0 +1,483 @@
+    "filename": "",
+    "runtime_image": "",
+    "outputs": [],
+    "env_vars": [],
+    "dependencies": [],
+    "include_subdirectories": False
+}
+
+
+def get_id_from_name(name):
+    return ' '.join(name.lower().replace('-', '').split()).replace(' ', '-').replace('_', '-')
+
+
+def set_node_type_data(id, label, description):
+    node_type = {
+        'id': "",
+        'op': f"execute-{id}-node",
+        'type': "execution_node",
+        'inputs': [inputs],
+        'outputs': [outputs],
+        'parameters': {},
+        'app_data': {
+            'ui-data': {
+                'label': label,
+                'description': description,
+                'image': "",
+                'x_pos': 0,
+                'y_pos': 0
+            }
+        }
+    }
+
+    return node_type
+
+
+class ComponentParser(SingletonConfigurable):
+    _type = "local"
+    properties: dict() = {}
+    parameters: dict() = {}
+
+    def __init__(self):
+        super().__init__()
+        self.properties = self.get_common_config('properties')
+
+    def get_common_config(self, config_name):
+        common_dir = os.path.join(os.path.dirname(__file__), 'resources')
+        common_file = os.path.join(common_dir, f"{config_name}.json")
+        with io.open(common_file, 'r', encoding='utf-8') as f:
+            common_json = json.load(f)
+
+        return common_json
+
+    def _get_component_catalog_json(self):
+        catalog_dir = os.path.join(os.path.dirname(__file__), "resources")
+        catalog_file = os.path.join(catalog_dir, f"{self._type}_component_catalog.json")
+        with open(catalog_file, 'r') as f:
+            catalog_json = json.load(f)
+
+        return catalog_json['components']
+
+    def list_all_components(self):
+        components = []
+        if self._type != "local":
+            for component in self._get_component_catalog_json():
+                assert "path" in component
+                components.append(component)
+
+        return components
+
+    def return_component_if_exists(self, component_id):
+        for component in self._get_component_catalog_json():
+            if component['id'] == component_id:
+                return component
+        return None
+
+    def add_component(self, request_body):
+        component_json = {
+            'name': request_body['name'],
+            'id': get_id_from_name(request_body['name']),
+            'path': request_body['path']
+        }
+
+        catalog_json = self._get_component_catalog_json()
+        catalog_json['components'].append(component_json)
+
+        catalog_dir = os.path.join(os.path.dirname(__file__), "resources")
+        catalog_file = os.path.join(catalog_dir, f"{self._type}_component_catalog.json")
+        with open(catalog_file, 'a') as f:
+            f.write(json.dumps(catalog_json))
+
+    def parse_component_details(self, component, component_name):
+        """Get component name, id, description for palette JSON"""
```

Comment:
Since this class isn't abstract, this should probably raise `NotImplementedError` if this method is required to be overwritten.  If it's optional, `pass` is fine.  

Same comment applies to `parse_component_properties()`